cmd/kubeadm/app/node/temp_util.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+
+// TODO: this whole file is just a temporary solution until the transition to k8s.io/client-go is finished.
+// It contains copies of the functions in kubeadm/app/cmd/util/kubeconfig.go that work with a non-versioned client
+// and should be removed in favor of the original kubeconfig.go file once the signature of csr.RequestNodeCertificate() is fixed.
+
+package node
+
+import (
+	"os"
+	"path"
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+)
+
+// TODO: remove this in favor of kubeadmutil.WriteKubeconfigIfNotExists() when csr.RequestNodeCertificate() is fixed
+func WriteKubeconfigIfNotExists(name string, kubeconfig *clientcmdapi.Config) error {
```

Comment:
@luxas I removed the `RequestNodeCertificate()` function that I had copied from `kubelet/util/csr` and went back to using an unversioned client for the CSR. However, since the CSR (and the client creation for the CSR) uses a bunch of utility functions from `kubeadm/app/util/kubeconfig.go`, I was forced to create copies again, because now those utilities functions are shared between places where a version client is used, and the CSR that uses a non-versioned one.

This is terribly ugly again, I cannot just change the signatures of the `kubeadmutil` functions just because the CSR uses an unversioned client, as those are used in other places too - that would mean going back to not using `client-go` anywhere.

WDYT? On purpose I have this change in a second commit in case we want to revert it ...